### ISSUE TITLE
Expose clang resource/include directory through CMake variable

### ISSUE THREAD

**Issue by tristan957** [llvm/llvm-project#58918]
I am using libclang to parse C source files. Given a small program like:

```c
#include <stddef.h>

int main(void)
{
  size_t hello = 5;
  return hello;
}
```

libclang should report `fatal error: 'stddef.h' file not found`.

On my system (Fedora 36), the resource directory is located:

```shell
$ clang -print-resource-dir
/usr/lib64/clang/14.0.5
```

Which means the include directory is located at `$(clang -print-resource-dir)/include)`. The clang cmake modules already expose a variable called `CLANG_INCLUDE_DIRS`, but the only directory exposed in this variable is `$CMAKE_INSTALL_PREFIX/include`, which isn't helpful for what I need.

Ideally these constants would be exposed in a cross buildsystem way like https://github.com/llvm/llvm-project/issues/9777, but I am willing to settle for just appending to the `CLANG_INCLUDE_DIRS` array since that should be as little effort as possible to solve what I need.

**Comment by efriedma-quic** [comment]
Normal usage patterns of libclang should normally find clang's resource directory without requiring you to explicitly pass it in.  If that isn't happening, something has gone wrong.  See CIndexer::getClangResourcesPath.

**Comment by tristan957** [comment]
Where is that documented in libclang? I don't see it any header file at my disposal?

**Comment by tristan957** [comment]
I tried switching to `clang_parseTranslationUnit2FullArgv()`, but that just has the same problems.

**Comment by tristan957** [comment]
Here is a command like which I pass to `clang_parse...()`.

```
clang -Ilib/libhse-3.so.3.1.0.p -Ilib -I../lib -Ilib/include -I../lib/include -Iinclude -I../include -Ilib/util/include -I../lib/util/include -Ilib/mpool/include -I../lib/mpool/include -Isubprojects/xxHash-0.8.1 -I../subprojects/xxHash-0.8.1 -Isubprojects/lz4-1.9.3/lib -I../subprojects/lz4-1.9.3/lib -Isubprojects/cJSON-1.7.15/include-workaround-meson -I../subprojects/cJSON-1.7.15/include-workaround-meson -Isubprojects/hyperloglog -I../subprojects/hyperloglog -Ilib/error/include -I../lib/error/include -Ilib/logging/include -I../lib/logging/include -Ilib/pidfile/include -I../lib/pidfile/include -Ilib/rest/include -I../lib/rest/include -Isubprojects/rbtree -I../subprojects/rbtree -Isubprojects/crc32c -I../subprojects/crc32c -Isubprojects/xoroshiro -I../subprojects/xoroshiro -fvisibility=hidden -fcolor-diagnostics -D_FILE_OFFSET_BITS=64 -O0 -g -fmacro-prefix-map=../= -DDEBUG_RCU -fstack-protector-all -DHSE_BUILD_DEBUG -D_GNU_SOURCE -D_LGPL_SOURCE -DLOG_DEFAULT=7 -DLEVEL1_DCACHE_LINESIZE=64 -DURCU_INLINE_SMALL_FUNCTIONS -fPIC -pthread -DXXH_INLINE_ALL -DXXH_STATIC_LINKING_ONLY -MD -MQ lib/libhse-3.so.3.1.0.p/util_lib_dax.c.o -MF lib/libhse-3.so.3.1.0.p/util_lib_dax.c.o.d -c -x c-header -std=gnu11 /home/tpartin/Projects/hse/lib/util/include/hse/util/dax.h -DHSE_MOCK=__attribute__((__annotate__("mock"))) -I /usr/lib64/clang/14.0.5/include
```

**Comment by efriedma-quic** [comment]
Looks like the relevant bits were last touched in https://reviews.llvm.org/D124815 .  CC @aykevl

**Comment by tristan957** [comment]
I appreciate your help up to this point with my issue :)

**Comment by aykevl** [comment]
@efriedma-quic true, but the patch should only have an effect when `dladdr` returns NULL - and thus only turn a crash into a non-crash. So I doubt it's related to this bug.
